test/tree.c

```diff
@@ -254,10 +258,96 @@ static void test_lookup_2(void)
 	nvme_free_tree(r);
 }
 
+bool test_src_addr()
```

Comment:
Thanks for the test. Could you split this patch into the function adding and the test? It makes the back porting simpler (in cases where we still don't have the test infrastructure backported :))